pages/api/socketio.js

```diff
@@ -1,66 +1,49 @@
 import { Server } from "socket.io";
 import { PIPE_API } from "../../config";
-import { connectToDatabase, insert, find } from "../../utils/nedb";
+import getHandle from "../../log_handler_getter";
+import handlePipeStructureMessage from "../../log_handlers/pipe_creation_handler"
+
 
 const ioHandler = async (req, res) => {
   if (!res.socket.server.io) {
-    const { db } = connectToDatabase();
-
     const io = new Server(res.socket.server, {
       maxHttpBufferSize: 1e8,
     });
 
     io.on("connection", (socket) => {
-
       socket.on("join_room", (msg) => {
         socket.join(msg);
       });
-      
-      socket.on('disconnect', function () {
-        console.log('disconnect client event....');
+
+      socket.on("disconnect", function () {
+        console.log("disconnect client event....");
       });
 
       socket.on("log", (msg) => {
         msg = JSON.parse(msg);
         msg["source"] = msg["source"].split(".").at(-1);
 
         let logType = msg["message"].split(":")[0];
-
-        if (
-          logType != "input" &&
-          logType != "output" &&
-          logType != "extra_image"
-        ) {
-          insert(db[msg["level"].toLowerCase() + "s"], msg);
-        } else {
-          let socketMessage = null;
-
-          if (logType == "extra_image") {
-            socketMessage = extractExtraImageDetailsFromExtraInputMessage(msg);
-          } else {
-            socketMessage = extractImageFromInputOutputMessage(msg);
-          }
-
-          socket.to(msg["source"]).emit(logType, socketMessage)
-        }
+        const handle = getHandle(logType);
```

Comment:
Did you mean:
```suggestion
        const handler = getHandler(logType);
```

Reply:
Handler it's an object that holds the "handle" method.
In that case we get the handle function and not the handle object because js doesn't like interfaces and we don't need to create those methods in objects..